static/app/views/performance/newTraceDetails/traceDrawer/details/span/sections/measurements.tsx

```diff
@@ -0,0 +1,110 @@
+import styled from '@emotion/styled';
+import type {Location} from 'history';
+
+import {
+  getFieldTypeFromUnit,
+  isNotMarkMeasurement,
+  isNotPerformanceScoreMeasurement,
+} from 'sentry/components/events/eventCustomPerformanceMetrics';
+import {t} from 'sentry/locale';
+import type {Organization} from 'sentry/types/organization';
+import {
+  DURATION_UNITS,
+  FIELD_FORMATTERS,
+  SIZE_UNITS,
+} from 'sentry/utils/discover/fieldRenderers';
+import {NumberContainer} from 'sentry/utils/discover/styles';
+import {isCustomMeasurement} from 'sentry/views/dashboards/utils';
+import type {TraceTree} from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
+import type {TraceTreeNode} from 'sentry/views/performance/newTraceDetails/traceModels/traceTreeNode';
+
+import {type SectionCardKeyValueList, TraceDrawerComponents} from '../../styles';
+import {TraceDrawerActionValueKind} from '../../utils';
+
+export function hasSpanMeasurements(span: TraceTree.Span) {
+  return !!span.measurements && Object.keys(span.measurements).length > 0;
+}
+
+function Measurements({
+  node,
+  location,
+  organization,
+}: {
+  location: Location;
+  node: TraceTreeNode<TraceTree.Span>;
+  organization: Organization;
+}) {
+  const {measurements} = node.value;
+
+  const measurementNames = measurements
+    ? Object.keys(measurements)
+        .filter(name => isCustomMeasurement(`measurements.${name}`))
+        .filter(isNotMarkMeasurement)
+        .filter(isNotPerformanceScoreMeasurement)
+        .sort()
+    : [];
+
+  if (measurementNames.length < 1) {
+    return null;
+  }
+
+  const items: SectionCardKeyValueList = [];
```

Comment:
This entire block may benefit from a `useMemo`

Reply:
Added `useMemo`